send_to_dock/test/test_send_to_dock_node.cpp

```diff
@@ -0,0 +1,131 @@
+  SendToDockNodeWrapper() {};
+
+  void HandleService(const std::shared_ptr<SetBoolSrv::Request> request,
+                     std::shared_ptr<SetBoolSrv::Response> response) {
+    return send_to_dock::SendToDockNode::HandleService(request, response);
+  }
+
+  void SetActiveGoal() {
+    send_to_dock::SendToDockNode::active_goal_ =
+        std::shared_ptr<ClientGoalHandleDockRobot>(
+            reinterpret_cast<ClientGoalHandleDockRobot *>(0x1), [](auto *) {});
+  }
+};
+
+class TestSendToDockNode : public testing::Test {
+protected:
+  TestSendToDockNode();
+  rclcpp_action::Server<DockRobot>::SharedPtr CreateDockServer();
+  std::shared_ptr<SendToDockNodeWrapper> node_;
+  std::shared_ptr<SetBoolSrv::Request> request_;
+  std::shared_ptr<SetBoolSrv::Response> response_;
+  rclcpp_action::Server<DockRobot>::SharedPtr action_server_;
+};
+
+TestSendToDockNode::TestSendToDockNode() {
+  node_ = std::make_shared<SendToDockNodeWrapper>();
+  request_ = std::make_shared<SetBoolSrv::Request>();
+  response_ = std::make_shared<SetBoolSrv::Response>();
+  action_server_ = nullptr;
+}
+
+rclcpp_action::Server<DockRobot>::SharedPtr
+TestSendToDockNode::CreateDockServer() {
+  return rclcpp_action::create_server<DockRobot>(
+      node_, "dock_robot",
+      // handle_goal
+      [](const rclcpp_action::GoalUUID &,
+         std::shared_ptr<const DockRobot::Goal>) {
+        return rclcpp_action::GoalResponse::ACCEPT_AND_EXECUTE;
+      },
+      // handle_cancel
+      [](const std::shared_ptr<ServerGoalHandleDockRobot>) {
+        return rclcpp_action::CancelResponse::ACCEPT;
+      },
+      // handle_accepted
+      [](const std::shared_ptr<ServerGoalHandleDockRobot> goal_handle) {
+        auto result = std::make_shared<DockRobot::Result>();
+        goal_handle->succeed(result);
+      });
+}
+
+TEST_F(TestSendToDockNode, DockServerIsNotCreated) {
+  node_->HandleService(request_, response_);
+  ASSERT_FALSE(response_->success);
+  EXPECT_EQ(response_->message, "Docking action server is not available");
+}
+
+TEST_F(TestSendToDockNode, DockServerCreatedAndTrueRequest) {
+  action_server_ = CreateDockServer();
+  request_->data = true;
+  node_->HandleService(request_, response_);
+  ASSERT_TRUE(response_->success);
+  EXPECT_EQ(response_->message, "New docking goal request sent.");
+}
+
+TEST_F(TestSendToDockNode, DockServerCreatedAndFalseRequest) {
+  action_server_ = CreateDockServer();
+  request_->data = false;
+  node_->HandleService(request_, response_);
+  ASSERT_TRUE(response_->success);
+  EXPECT_EQ(response_->message, "No active docking goal to cancel.");
+}
+
+TEST_F(TestSendToDockNode, DockRobotTrueRequestAndExistingGoal) {
+  action_server_ = CreateDockServer();
+  node_->SetActiveGoal();
+  request_->data = true;
+  node_->HandleService(request_, response_);
+  ASSERT_TRUE(response_->success);
+  EXPECT_EQ(response_->message, "Docking goal already exists.");
+}
+
+TEST_F(TestSendToDockNode, DockRobotFalseRequestAndExistingGoal) {
```

Comment:
```suggestion  
TEST_F(TestSendToDockNode, CheckIfSendToDockCancelsExistingGoal) {
```

We can add checking if the active goal is `nullptr`.